pwnlib/dynelf.py

```diff
@@ -823,6 +824,7 @@ def _resolve_symbol_gnu(self, libbase, symb, hshtab, strtab, symtab):
                 # Check for collision on hash values
                 sym  = symtab + sizeof(Sym) * (ndx + i)
                 name = leak.s(strtab + leak.field(sym, Sym.st_name))
+                name = name.decode('utf-8')
 
                 if name == symb:
```

Comment:
I would prefer conversion of `symb` to bytes here, rather than `name` to be a string.  WDYT @Arusekk?

Reply:
Hi, just to add my 2 cents.

Nice spot! That would make sense and be safer :+1: 

However, if `name` did not contain ASCII, then its indicative something much more serious has gone wrong - memory corruption/bad geometry calculations and its unlikely that DynELF will work at all. So maybe, crashing here is not such a problem?

Reply:
Ideally we should not crash at all ;)

Sticking with raw byte manipulation seems like the best route.  If the user wants to search for "🙃.so" it's on them to encode that into the correct byte format.

Reply:
I think that `symb` should be converted to bytes, and permitting bytes (not only string) in the first place.